joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/ClassTests.scala

```diff
@@ -389,4 +390,75 @@ class ClassTests extends RubyCode2CpgFixture {
       }
     }
   }
+
+  "Instance variable in class" should {
+    val cpg = code("""
+        |class Foo
+        | @a
+        |
+        | def foo
+        |   @b = 10
+        | end
+        |
+        | def foobar
+        |   @c = 20
+        |   @d = 40
+        | end
+        |
+        | def barfoo
+        |   puts @a
+        |   puts @c
+        |   @o = "a"
+        | end
+        |end
+        |""".stripMargin)
+
+    "be correct" in {
+      inside(cpg.typeDecl.name("Foo").l) {
+        case fooType :: Nil =>
+          inside(fooType.member.l) {
+            case aMember :: bMember :: cMember :: dMember :: oMember :: Nil =>
+              // Test that all members in class are present
+              aMember.code shouldBe "@a"
+              bMember.code shouldBe "@b"
+              cMember.code shouldBe "@c"
+              dMember.code shouldBe "@d"
+              oMember.code shouldBe "@o"
+            case _ => fail("Expected 5 members")
+          }
+
+          inside(fooType.method.name(Defines.ConstructorMethodName).l) {
```

Comment:
The block below could be under a different heading like `"create nil assignments under the class initializer" in {`